<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
    <head>
        <meta http-equiv="Content-Type" content="text/html;charset=utf-8" ></meta>
        <title>scala/xml/include/XIncludeException.scala</title>
        <script type="text/javascript" src="../../../jquery-all.js"></script>
        <script type="text/javascript" src="../../../linked.js"></script>
        <link rel="stylesheet" type="text/css" href="../../../style.css" title="Style"></link>
    </head>
    <body>
        <pre>
<span class="comment">/*                     __                                               *\
**     ________ ___   / /  ___     Scala API                            **
**    / __/ __// _ | / /  / _ |    (c) 2002-2013, LAMP/EPFL             **
**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/               **
** /____/\___/_/ |_/____/_/ | |                                         **
**                          |/                                          **
\*                                                                      */</span>

package scala
package xml
package include

<span class="comment">/**
 * `XIncludeException` is the generic superclass for all checked exceptions
 * that may be thrown as a result of a violation of XInclude's rules.
 *
 * Constructs an `XIncludeException` with the specified detail message.
 * The error message string `message` can later be retrieved by the
 * `{@link java.lang.Throwable#getMessage}`
 * method of class `java.lang.Throwable`.
 *
 * @param   message   the detail message.
 */</span>
class <a title="class XIncludeException extends Exception" id="scala.xml.include;XIncludeException">XIncludeException</a><a href="#scala.xml.include;XIncludeException" title="scala.xml.include.XIncludeException" class="delimiter">(</a><a title="String" id="scala.xml.include;XIncludeException.message">message</a>: <span title="String">String</span><span class="delimiter">)</span> extends <a href="../../package.scala.html#scala.package;Exception" title="Exception">Exception</a><span class="delimiter">(</span><a href="#scala.xml.include;XIncludeException.message" title="String">message</a><span class="delimiter">)</span> <span class="delimiter">{</span>

  <span class="comment">/**
   * uses `'''null'''` as its error detail message.
   */</span>
  def this<span class="delimiter">(</span><span class="delimiter">)</span> = this<span class="delimiter">(</span>null<span class="delimiter">)</span>

  private var <a title="Throwable" id="scala.xml.include;XIncludeException.rootCause_=">rootCause</a>: <span title="Throwable">Throwable</span> = null

  <span class="comment">/**
   * When an `IOException`, `MalformedURLException` or other generic
   * exception is thrown while processing an XML document for XIncludes,
   * it is customarily replaced by some form of `XIncludeException`.
   * This method allows you to store the original exception.
   *
   * @param   nestedException   the underlying exception which
   * caused the XIncludeException to be thrown
   */</span>
  def <a title="(nestedException: Throwable)Unit" id="scala.xml.include;XIncludeException.setRootCause">setRootCause</a><span class="delimiter">(</span><a title="Throwable" id="scala.xml.include;XIncludeException.setRootCause.nestedException">nestedException</a>: <span title="Throwable">Throwable</span><span class="delimiter">)</span> <span class="delimiter">{</span>
    this.<a href="#scala.xml.include;XIncludeException.rootCause_=" title="(x$1: Throwable)Unit">rootCause</a> = <a href="#scala.xml.include;XIncludeException.setRootCause.nestedException" title="Throwable">nestedException</a>
  <span class="delimiter">}</span>

  <span class="comment">/**
   * When an `IOException`, `MalformedURLException` or other generic
   * exception is thrown while processing an XML document for XIncludes,
   * it is customarily replaced by some form of `XIncludeException`.
   * This method allows you to retrieve the original exception.
   * It returns null if no such exception caused this `XIncludeException`.
   *
   * @return Throwable   the underlying exception which caused the
   *                     `XIncludeException` to be thrown
   */</span>
  def <a title="()Throwable" id="scala.xml.include;XIncludeException.getRootCause">getRootCause</a><span class="delimiter">(</span><span class="delimiter">)</span>: <span title="Throwable">Throwable</span> = this.<a href="#scala.xml.include;XIncludeException.rootCause_=" title="=&gt; Throwable">rootCause</a>

<span class="delimiter">}</span>

        </pre>
    </body>
</html>
